### ISSUE TITLE
Repo is very large

### ISSUE THREAD

**Issue by martinetd** [cawilliamson/ansible-gpdpocket#3]
Please consider removing (and filter-branch'ing out) the kernel binaries from this repo, as they were before.

I understand it is simpler when you point out to people a single zip release, but I want to clone this repo and update it once in a while to see if there have been new tricks -- I doubt I'm the only one in that case.
Wasting 50ish MB of bandwidth for every kernel update is already annoying now (555MB as of today) and this repo is barely one week old (especially because I do not want to use it, but even if I had you only need to grab one whenever you want to run the playbooks again, and will not need every single intermediate updates)

Now the moaning is out of the way, thank you for gathering all these fixes and the time spent there.
Feel free to ignore/reject this but I'll just never update :P

**Comment by cawilliamson** [comment]
I agree that the repo has gotten very unwieldy size wise and I have now flattened the repo to remove all previous large binaries which are being pulled.

As for the kernel updates - this no longer an issue:

- There is now only a single kernel being bundled with this repository - that's the bootstrap kernel, used for first install and iso bootstrapping.

- This kernel can be used as a recovery kernel (it will never be removed) which helps with the next point.

- All kernel updates are now pulled *directly* from the repository at github.com/jwrdegoede/linux-sunxi as needed. A local copy of this repo will be kept on your system and a pull will happen only when there are changes.

So in short - this approach has the following benefits:

- You only pull the ~50MB bootstrap kernel once and install it once - ever. After that - your `git pull` wont re-download. In order to update your system you simply run `sudo gpd-update` - that's really all there is to it - all of this stuff will be maintained automatically for you.

- You pull the source code for the kernel directly from source instead of having a delay and build it yourself meaning: faster releases and complete transparency about what is in your kernel (since you're building it.)

Does this solve your concern? :)

**Comment by martinetd** [comment]
Great. Thanks! I can confirm a fresh clone is much smaller. git gc will take care of my current clone one of these days.

I'd have considered slightly more subtle with filter-branch, but if you're fine with flattening (as you just have) that works for me.
I'm way too paranoid for the other points to concern me much, but it sounds fine to me :)

**Comment by cawilliamson** [comment]
I'll look in to git filter-branch - not something I've used before to be honest!

The history on this repo was a mess though (lots of binaries uploaded and removed later) so a clean out was in order.

Thanks for the feedback - much appreciated!
